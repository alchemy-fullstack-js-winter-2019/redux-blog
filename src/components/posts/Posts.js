import React from 'react';
import PropTypes from 'prop-types';
import Post from './Post';
import { Link } from 'react-router-dom';

export default function Posts({ posts }) {
  const listOfPosts = posts.map((post) => {
    console.log('post id', post.id);
    return (
      <Link to={`/posts/${post.id}`} key={post.id}>
        <li >
          <Post post={post} />
        </li>
      </Link>
    );
  });

  return (
    <>
      <ol>
        {listOfPosts}
      </ol>
    </>
  );
}

Posts.propTypes = {
  posts: PropTypes.array
};
