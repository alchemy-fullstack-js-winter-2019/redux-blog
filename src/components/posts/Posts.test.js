import React from 'react';
import renderer from 'react-test-renderer';
import Posts from './Posts';
import { MemoryRouter } from 'react-router-dom';

describe('Posts Snapshot', () => {
  it('matches a snapshot', () => {
    const postsTest = [    
      {
        'userId': 1,
        'id': 1,
        'title': 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
        'body': 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'
      },
      {
        'userId': 1,
        'id': 2,
        'title': 'qui est esse',
        'body': 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla'
      }];
    const tree = renderer.create(
      <MemoryRouter>
        <Posts posts={postsTest} term='term' onChange={jest.fn()}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
