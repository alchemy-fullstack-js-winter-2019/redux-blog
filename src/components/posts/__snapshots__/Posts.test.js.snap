// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Snapshot matches a snapshot 1`] = `
Array [
  <input
    name="term"
    onChange={[MockFunction]}
    type="text"
    value="term"
  />,
  <ul>
    <a
      href="/posts/1"
      onClick={[Function]}
    >
      <li>
        sunt aut facere repellat provident occaecati excepturi optio reprehenderit
      </li>
    </a>
    <a
      href="/posts/2"
      onClick={[Function]}
    >
      <li>
        qui est esse
      </li>
    </a>
  </ul>,
]
`;
